src/scipp/core/__init__.py

```diff
@@ -57,27 +61,27 @@
 
 from .deprecation import _deprecated_attrs, _deprecated_meta, _deprecated_drop_attrs
 
-DataArray.attrs = property(_deprecated_attrs)
-DataArray.meta = property(_deprecated_meta)
-DataArray.drop_attrs = _deprecated_drop_attrs
+DataArray.attrs = property(_deprecated_attrs)  # type: ignore[assignment, method-assign]
+DataArray.meta = property(_deprecated_meta)  # type: ignore[assignment, method-assign]
+DataArray.drop_attrs = _deprecated_drop_attrs  # type: ignore[assignment, method-assign]
```

Comment:
Mypy doesn't like this because `cpp_classes.pyi` lists all methods of `Dataarray`, etc. and we are not allowed to assign to methods. However, at runtime, these methods don't exist before the assignments here. So we just need to silence mypy and make it use the stubs instead.